### ISSUE TITLE
how to solve the overfitting problem

### ISSUE THREAD

**Issue by linhanxiao** [facebookresearch/fairseq-py#78]
i have a new dataset，when i trained on this new dataset，the valid loss increased  and the train loss decreased，this caused the overfitting problem，I have added the L2 regularization ,but it didn't help,so what else should i do to solve this problem @ynd @jgehring @

**Comment by linhanxiao** [comment]
| epoch 001 | train loss 740.61 | train ppl 11111111.00 | s/checkpoint      61 | words/s   1288 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 918.1928
| epoch 001 | valid on 'valid' subset | valid loss 7.03 | valid ppl 111111111.00                                                                
| epoch 002 | train loss 6.08 | train ppl 11111111.00 | s/checkpoint      59 | words/s   1309 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.7315
| epoch 002 | valid on 'valid' subset | valid loss 5.79 | valid ppl 111111111.00                                                                
| epoch 003 | train loss 5.34 | train ppl 11111111.00 | s/checkpoint      59 | words/s   1329 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.6352
| epoch 003 | valid on 'valid' subset | valid loss 5.53 | valid ppl 111111111.00                                                                
| epoch 004 | train loss 5.00 | train ppl 11111111.00 | s/checkpoint      60 | words/s   1307 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.6491
| epoch 004 | valid on 'valid' subset | valid loss 5.41 | valid ppl 111111111.00                                                                
| epoch 005 | train loss 4.75 | train ppl 11111111.00 | s/checkpoint      60 | words/s   1296 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.5962
| epoch 005 | valid on 'valid' subset | valid loss 5.61 | valid ppl 111111111.00                                                                
| epoch 006 | train loss 4.50 | train ppl 11111111.00 | s/checkpoint      60 | words/s   1303 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.5554
| epoch 006 | valid on 'valid' subset | valid loss 5.39 | valid ppl 111111111.00                                                                
| epoch 007 | train loss 4.29 | train ppl 11111111.00 | s/checkpoint      60 | words/s   1296 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.5964
| epoch 007 | valid on 'valid' subset | valid loss 5.37 | valid ppl 111111111.00                                                                
| epoch 008 | train loss 4.14 | train ppl 11111111.00 | s/checkpoint      59 | words/s   1309 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.7772
| epoch 008 | valid on 'valid' subset | valid loss 5.41 | valid ppl 111111111.00                                                                
| epoch 009 | train loss 3.96 | train ppl 11111111.00 | s/checkpoint      60 | words/s   1302 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.7252
| epoch 009 | valid on 'valid' subset | valid loss 5.50 | valid ppl 111111111.00                                                                
| epoch 010 | train loss 3.85 | train ppl 11111111.00 | s/checkpoint      60 | words/s   1308 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.8596
| epoch 010 | valid on 'valid' subset | valid loss 5.48 | valid ppl 111111111.00                                                                
| epoch 011 | train loss 3.72 | train ppl 11111111.00 | s/checkpoint      59 | words/s   1313 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.8970
| epoch 011 | valid on 'valid' subset | valid loss 5.65 | valid ppl 111111111.00                                                                
| epoch 012 | train loss 3.59 | train ppl 11111111.00 | s/checkpoint      59 | words/s   1312 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.7799
| epoch 012 | valid on 'valid' subset | valid loss 5.67 | valid ppl 111111111.00                                                                
| epoch 013 | train loss 3.48 | train ppl 11111111.00 | s/checkpoint      59 | words/s   1316 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 1.2961
| epoch 013 | valid on 'valid' subset | valid loss 5.67 | valid ppl 111111111.00                                                                
| epoch 014 | train loss 3.39 | train ppl 11111111.00 | s/checkpoint      59 | words/s   1313 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.8846
| epoch 014 | valid on 'valid' subset | valid loss 5.73 | valid ppl 111111111.00                                                                
| epoch 015 | train loss 3.32 | train ppl 11111111.00 | s/checkpoint      59 | words/s   1310 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.8443
| epoch 015 | valid on 'valid' subset | valid loss 5.91 | valid ppl 111111111.00                                                                
| epoch 016 | train loss 3.26 | train ppl 11111111.00 | s/checkpoint      60 | words/s   1306 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.7615
| epoch 016 | valid on 'valid' subset | valid loss 5.85 | valid ppl 111111111.00                                                                
| epoch 017 | train loss 3.15 | train ppl 11111111.00 | s/checkpoint      59 | words/s   1310 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.9267
| epoch 017 | valid on 'valid' subset | valid loss 6.01 | valid ppl 111111111.00                                                                
| epoch 018 | train loss 3.13 | train ppl 11111111.00 | s/checkpoint      60 | words/s   1306 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.8545
| epoch 018 | valid on 'valid' subset | valid loss 5.95 | valid ppl 111111111.00                                                                
| epoch 019 | train loss 3.04 | train ppl 11111111.00 | s/checkpoint      59 | words/s   1328 | words/batch    838 | bsz    64 | lr 0.330000 | clip 100% | gnorm 0.9952
| epoch 019 | valid on 'valid' subset | valid loss 6.02 | valid ppl 11111111

**Comment by linhanxiao** [comment]
in the train.py , I didn't use the code'  lr = trainer.lr_step(val_loss, epoch)' @myleott

**Comment by Zrachel** [comment]
You must have modified some code, there shows an obvious bug that, 
valid ppl =111111111.00
valid ppl should be equal to pow(2, loss)

**Comment by linhanxiao** [comment]
yes  i change the code,i set the ppl =111111111.00, @Zrachel

**Comment by linhanxiao** [comment]
i want the valid loss to decrease when the train loss is decreasing @Zrachel

**Comment by Zrachel** [comment]
You didn't provide your data, you didn't provide your modified code. I don't think anyone could help you.

Moreover, as overfitting is such a common problem, I think search on google first is better :) The authors of convs2s is not responsible for tackle these.

**Comment by linhanxiao** [comment]
I have searched  on google，and i have added the L2 regularization to the loss function but it doesn't  help ,what else should i do for solving overfitting problem ,i need some advices @Zrachel

**Comment by huihuifan** [comment]
How large is your new dataset?

**Comment by Zrachel** [comment]
Try to decrease the parameter size, for example, set the number of convolution layers to one.

**Comment by linhanxiao** [comment]
@huihuifan i only have 6000 example for trainset ,1000example for valid set ,1000 example for test set

**Comment by linhanxiao** [comment]
thank you for your advice @Zrachel ,i am wondering why the L2 regularization didn't help to solve the overfitting problem

**Comment by frankang** [comment]
6000 examples for the training dataset is too small. Please try a larger dataset, i.e., 600,000 sentence pairs for training.

**Comment by linhanxiao** [comment]
what about 80000 examples for training dataset? @frankang  and what if i only have 6000 example?

**Comment by huihuifan** [comment]
Your dataset is too small. You can train on a larger dataset and fine-tune on your smaller dataset.

**Comment by linhanxiao** [comment]
thank you for your advice!! @huihuifan

**Comment by huihuifan** [comment]
Closing this issue.
